packages/auth-server/src/adapter/prisma-client.js

```diff
@@ -0,0 +1,15 @@
+import createClient from '@pins/database/src/create-client.js';
+
+/** @type {import('@pins/database/src/create-client').prismaConfig} */
+const prismaConfig = {
+	datasourceUrl: process.env.SQL_CONNECTION_STRING
```

Comment:
Ooh here's a scab to pick. I have liked having the config objects because you get the inferred type and autocomplete _but_ we could just type the process by having a declaration file like this

```
/** Defines the typings for `process.env.VARIABLES` */
export interface Climate360ProcessEnv {
  NEXT_PUBLIC_BACKEND_BASE_URL: string
}

declare global {
  // eslint-disable-next-line @typescript-eslint/no-namespace
  namespace NodeJS {
    // https://github.com/typescript-eslint/typescript-eslint/issues/1640
    // eslint-disable-next-line @typescript-eslint/no-empty-interface
    interface ProcessEnv extends Climate360ProcessEnv {
      // add members to Climate360ProcessEnv, not here
    }
  }
}
```

I think it's also worth introducing something like this 

```
/** Checks whether the environment variables conform to expectations */
export function checkEnv(): void {
  const boolStringAllowedValues = ["true", "false"];
  const nonEmptyRegex = /.+/;
  const pathRegex = /^\/.*/;

  checkRegex("API_KEY", nonEmptyRegex);
  checkString("ASSET_PREFIX", true);
  checkRegex("CI_VERSION", nonEmptyRegex);
  checkType("CONTENT_SECURITY_POLICY", ["local", "secure"]);
  checkRegex("HOST_ENV", nonEmptyRegex);
  // dev-only env vars
  checkRegex("API_PROXY_PATH", pathRegex, true);
  checkURL("API_PROXY_URL", true);
  checkType("DISABLE_APPD", boolStringAllowedValues, true);
  checkType("ENABLE_API_PROXY", boolStringAllowedValues, true);
  checkType("USE_LOCAL_SSL", boolStringAllowedValues, true);
  checkType("SILENCE_WINSTON", boolStringAllowedValues, true);
  checkString("APPDYNAMICS_ACCESS_KEY");
}
```

to run as the app boots so we know it's configured correctlyish

Reply:
We use `joi` on the back office to check config values:
https://github.com/Planning-Inspectorate/back-office/blob/main/apps/api/src/server/config/schema.js#L3

Reply:
looking into the joi implementation, seems to log redacted secrets on failure which probably isn't ideal

Reply:
this is done